tests/system/model_monitoring/test_app.py

```diff
@@ -386,6 +386,58 @@ def custom_setup_class(cls) -> None:
     def custom_setup(cls) -> None:
         _V3IORecordsChecker.custom_setup(project_name=cls.project_name)
 
+    @classmethod
+    def custom_teardown(self):
+        # validate that stream resources were deleted as expected
+        stream_path = self._test_env[
+            "MLRUN_MODEL_ENDPOINT_MONITORING__STREAM_CONNECTION"
+        ]
+
+        func_to_validate = [
+            "model-monitoring-writer",
+            "histogram-data-drift",
+            "evidently-app-test-v2",
+        ]
+
+        if stream_path.startswith("v3io:///"):
+            from v3io.dataplane.response import HttpResponseError
+
+            from mlrun.utils.v3io_clients import get_v3io_client
+
+            client = get_v3io_client(endpoint=mlrun.mlconf.v3io_api)
+
+            for func in func_to_validate:
+                with pytest.raises(HttpResponseError):
+                    client.object.get(
+                        container="projects",
+                        path=f"{self.project_name}/model-endpoints/stream-{func}/serving-state.json",
+                    )
+
+            # validate that the monitoring stream was deleted
+            with pytest.raises(HttpResponseError):
+                client.object.get(
+                    container="projects",
+                    path=f"{self.project_name}/model-endpoints/stream/serving-state.json",
+                )
```

Comment:
I would even unify these two checks

Reply:
the stream of the "stream pod" is represents by empty string so it's kind of an overkill to put it in the same pattern. If we will do so, we will need to rename it afterwards in both v3io (`"{self.project_name}/model-endpoints/stream-/serving-state.json"` -> `{self.project_name}/model-endpoints/stream/serving-state.json"`) and kafka (`monitoring_stream_{self.project_name}_` -> `monitoring_stream_{self.project_name}`). 
Therefore, I prefer to keep it separately.